### ISSUE TITLE
Separate access control to plugin functions from built in functions

### ISSUE THREAD

**Issue by homar** [trinodb/trino#24435]
Currently built in functions and [plugin functions](https://github.com/trinodb/trino/blob/90db2f54f25c8b52db56bdfcabea2c1b4fa6dac9/core/trino-spi/src/main/java/io/trino/spi/Plugin.java#L65) all end up in `system.builtin` and access control treats them the same - they are always allowed. 

I believe it would be beneficial to allow more fine grained control  around plugin functions.

That would definitely require changes in file based access control and probably in other places that I am not aware of.

**Comment by martint** [comment]
This is intentional, since legacy plugin functions are not namespaced. To control access to plugin functions, you need to implement them as catalog functions using the newer APIs:  ConnectorMetadata.getFunctions(), etc.

**Comment by electrum** [comment]
I don't think that "legacy" is the right term. Catalog functions is a low level interface. There's no way to do annotation based functions using that API. If we want to deprecate the old way, then we need a clear migration path.

**Comment by lordicecream** [comment]
Duplicate of #19912

**Comment by mosabua** [comment]
See discussion in https://github.com/trinodb/trino/pull/26041
